### PR TITLE
fix: verifyBearerAuth to call done(error) on error

### LINKED ISSUE #90: When used with fastify-auth, if the bearer auth fails, no subsequent fastify-auth options execute
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure it has not already been reported

### Fastify version

3

### Plugin version

5.1.0

### Node.js version

14

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

11.2.3

### Description

When `fastify-bearer-auth` is registered with `{ addHook: false }`, so it can be used with `fastify-auth`, `fastify.verifyBearerAuth` needs to be the last element on `fastify.auth([])` or any other options after it will new be tested (because it responds immediately on failure - `reply.send()`).

Regarding errors it's also not ideal because, since it needs to be in the end of the lis, the errored response will always be from `fastify-bearer-auth`.

### Steps to Reproduce

Just add a preHandler hook similar to:

```
fastify.addHook('preHandler',
  fastify.auth([
    fastify.someAuth1, 
    fastify.verifyBearerAuth,
    fastify.someAuth2 // this will never be tested/executed
  ])
)
```

### Expected Behavior

For `fastify.verifyBearerAuth` not to respond back and to throw an error, which would make `fastify-auth` test the next handler in the list.

### PR DESCRIPTION
**Pull request by ahgentil**
instead of replying to client

close #90

#### Checklist

- [ x ] run `npm run test` and `npm run benchmark`
- [ x ] tests and/or benchmarks are included
- [ x ] documentation is changed or added
- [ x ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test/integration.test.js [hunk 1]
```diff
@@ -119,7 +119,81 @@ test('integration with fastify-auth', async (t) => {
...
+        url: '/bearer-first',
+        headers: {
+          authorization: 'Bearer fail'
+        }
+      })
+      t.equal(res.statusCode, 200)
+      t.match(JSON.parse(res.body).hello, 'world')
+    } catch (err) {
+      t.error(err)
+    }
```
Comment (mcollina) [inline test/integration.test.js:184]: All those try/catch are not needed.

**Review by mcollina — COMMENTED** [review]
good work!

File: Readme.md [hunk 2]
```diff
@@ -109,6 +109,11 @@ fastify.route({
 })
 ```
 
+By passing `{ addHook: false }` in the options, the `verifyBearerAuth` hook, instead of
+immediately replying on error (`reply.send(someError)`), it calls `done(someError)`. This
```
Comment (jsumners) [inline Readme.md:113]:
```suggestion
immediately replying on error (`reply.send(someError)`), invokes `done(someError)`. This
```

File: Readme.md [hunk 3]
```diff
@@ -109,6 +109,11 @@ fastify.route({
...
+will allow `fastify.auth` to take back control and try to authenticate the request with the
```
Comment (jsumners) [inline Readme.md:114]:
```suggestion
will allow `fastify.auth` to continue with the next authentication scheme in the hook list.
```

File: Readme.md [hunk 4]
```diff
@@ -109,6 +109,11 @@ fastify.route({
...
+next hook in the list. If `verifyBearerAuth` is the last hook in the list, `fastify.auth`
+will reply with `Unauthorized`.
```
Comment (jsumners) [inline Readme.md:116]:
```suggestion
If `verifyBearerAuth` is the last hook in the list, `fastify.auth` will reply with `Unauthorized`.
```

File: test/integration.test.js (hunk 1 above)
Comment (ahgentil) [reply test/integration.test.js]: you're right. done. thanks

File: Readme.md (hunk 2 above)
Comment (ahgentil) [reply Readme.md]: thanks for the suggestions.

**Review by mcollina — APPROVED** [review]
lgtm

@jsumners wdyt, minor or major?

**Review by jsumners — APPROVED** [review]

**Comment by jsumners** [comment]
Definitely a major.

**Pull request merged**